src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1996,6 +2005,41 @@ private static NullableAnnotation GetNullableAnnotation(BoundExpression expr)
             }
         }
 
+        private static TypeSymbolWithAnnotations GetNullCoalescingResultType(BoundExpression leftOperand, TypeSymbolWithAnnotations leftResult, BoundExpression rightOperand, TypeSymbolWithAnnotations rightResult, TypeSymbol resultType)
```

Comment:
>GetNullCoalescingResultType [](start = 49, length = 27)

Is it possible to assert that ```leftResult```, ```rightResult``` and ```resultType``` are all equal ignoring nullablility?

Reply:
No, because null-coalescing will pass in the unconverted type. See the `object v16 = u16 ?? new object();` part of `Conversions_01` for an example.

---
In reply to: [248489248](https://github.com/dotnet/roslyn/pull/32225#discussion_r248489248) [](ancestors = 248489248)

Reply:
> No, because null-coalescing will pass in the unconverted type. See the object v16 = u16 ?? new object(); part of Conversions_01 for an example.

This doesn't feel right. Shouldn't nullability of the converted value be taken into consideration? Can't it change by a user-defined conversion, for example? 

---
In reply to: [249135197](https://github.com/dotnet/roslyn/pull/32225#discussion_r249135197) [](ancestors = 249135197,248489248)